etna/core/mixins.py

```diff
@@ -89,6 +90,36 @@ def to_dict(self):
         params["_target_"] = BaseMixin._get_target_from_class(self)
         return params
 
+    def set_params(self, **params: dict) -> "BaseMixin":
+        """Return new object instance with modified parameters.
+
+        The method works on simple estimators as well as on nested objects
+        (such as :class:`~etna.pipeline.Pipeline`). The latter have
+        parameters of the form ``<component>.<parameter>`` so that it's
+        possible to update each component of a nested object.
+
+        Parameters
+        ----------
+        **params: dict
+            Estimator parameters.
+
+        """
+        params_dict = self.to_dict()
+
+        # we need to change specification into list of nested params, "model.depth" to nested structure
+        # the code below updates public_params by params
+        for param, param_value in params.items():
+            *param_nesting, param_attr = param.split(".")
+            cycle_dict = params_dict
+            for param_nested in param_nesting:
+                if cycle_dict.get(param_nested, -1) == -1:
```

Comment:
It looks like you can use `cycle_dict = cycle_dict.setdefault(param_nested, {})` instead of lines 115-117.